lib/CModHandler.h

```diff
@@ -253,17 +253,20 @@ class DLL_LINKAGE CModHandler
 
 		int CREEP_SIZE; // neutral stacks won't grow beyond this number
 		int WEEKLY_GROWTH; //percent
-		int NEUTRAL_STACK_EXP; 
+		int NEUTRAL_STACK_EXP;
 		int MAX_BUILDING_PER_TURN;
 		bool DWELLINGS_ACCUMULATE_CREATURES;
 		bool ALL_CREATURES_GET_DOUBLE_MONTHS;
 		int MAX_HEROES_AVAILABLE_PER_PLAYER;
 		int MAX_HEROES_ON_MAP_PER_PLAYER;
+		bool WINNING_HERO_WITH_NO_TROOPS_RETREATS;
 
 		template <typename Handler> void serialize(Handler &h, const int version)
 		{
 			h & data & CREEP_SIZE & WEEKLY_GROWTH & NEUTRAL_STACK_EXP & MAX_BUILDING_PER_TURN;
-			h & DWELLINGS_ACCUMULATE_CREATURES & ALL_CREATURES_GET_DOUBLE_MONTHS & MAX_HEROES_AVAILABLE_PER_PLAYER & MAX_HEROES_ON_MAP_PER_PLAYER;
+			h & DWELLINGS_ACCUMULATE_CREATURES & ALL_CREATURES_GET_DOUBLE_MONTHS &
+			MAX_HEROES_AVAILABLE_PER_PLAYER & MAX_HEROES_ON_MAP_PER_PLAYER &
+			WINNING_HERO_WITH_NO_TROOPS_RETREATS;

```

Comment:
"const int version" parameter is current used version. 

in Connection.h there are definitions of supported versions:

```cpp
const ui32 version = 755;
const ui32 minSupportedVersion = 753;
```

change it to 

```cpp
const ui32 version = 756;
const ui32 minSupportedVersion = 753;
```

and in serialization

```cpp
if(version >= 756) //save format backward compatibility
{
	h & WINNING_HERO_WITH_NO_TROOPS_RETREATS;
}
else if(!h.saving)
{
	WINNING_HERO_WITH_NO_TROOPS_RETREATS = false;
}
```